osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/config/SoapClientConfig.java

```diff
@@ -0,0 +1,126 @@
+    private Resource soapRsaPrivateKeyResource;
+
+    @Bean
+    Jaxb2Marshaller soapClientJaxb2Marshaller() {
+        Jaxb2Marshaller jaxb2Marshaller = new Jaxb2Marshaller();
+        jaxb2Marshaller.setContextPath(XSD_SCHEMA_PACKAGE);
+        return jaxb2Marshaller;
+    }
+
+    @Bean
+    public WebServiceTemplate webServiceTemplate() throws Exception {
+        WebServiceTemplate webServiceTemplate = new WebServiceTemplate();
+        webServiceTemplate.setMarshaller(soapClientJaxb2Marshaller());
+        webServiceTemplate.setUnmarshaller(soapClientJaxb2Marshaller());
+        webServiceTemplate.setDefaultUri(defaultUri);
+
+        if (Boolean.parseBoolean(useClientAuth)) {
+            webServiceTemplate.setMessageSender(httpsUrlConnectionMessageSender());
+        }
+
+        return webServiceTemplate;
+    }
+
+    @Bean
+    public HttpsUrlConnectionMessageSender httpsUrlConnectionMessageSender() throws Exception {
+        HttpsUrlConnectionMessageSender httpsUrlConnectionMessageSender =
+                new HttpsUrlConnectionMessageSender();
+        // set the trust store(s)
+        httpsUrlConnectionMessageSender.setTrustManagers(trustManagersFactoryBean().getObject());
+        // set the key store(s)
+        httpsUrlConnectionMessageSender.setKeyManagers(keyManagersFactoryBean().getObject());
+
+        return httpsUrlConnectionMessageSender;
+    }
+
+    @Bean
+    public KeyStoreFactoryBean trustStore() {
+        KeyStoreFactoryBean keyStoreFactoryBean = new KeyStoreFactoryBean();
+        keyStoreFactoryBean.setLocation(trustStore);
+        keyStoreFactoryBean.setPassword(trustStorePassword);
+
+        return keyStoreFactoryBean;
+    }
+
+    @Bean
+    public TrustManagersFactoryBean trustManagersFactoryBean() {
+        TrustManagersFactoryBean trustManagersFactoryBean = new TrustManagersFactoryBean();
+        trustManagersFactoryBean.setKeyStore(trustStore().getObject());
+
+        return trustManagersFactoryBean;
+    }
+
+    @Bean
+    public KeyStoreFactoryBean keyStore() {
+        KeyStoreFactoryBean keyStoreFactoryBean = new KeyStoreFactoryBean();
+        keyStoreFactoryBean.setLocation(keyStore);
+        keyStoreFactoryBean.setPassword(keyStorePassword);
+
+        return keyStoreFactoryBean;
+    }
+
+    @Bean
+    public KeyManagersFactoryBean keyManagersFactoryBean() {
+        KeyManagersFactoryBean keyManagersFactoryBean = new KeyManagersFactoryBean();
+        keyManagersFactoryBean.setKeyStore(keyStore().getObject());
+        // set the password of the key pair to be used
+        keyManagersFactoryBean.setPassword(keyPassword);
+
+        return keyManagersFactoryBean;
+    }
+
+    @Bean
+    public RsaEncryptionProvider rsaEncryptionProvider() {
+        try {
+            File privateRsaKeyFile = this.soapRsaPrivateKeyResource.getFile();
+            return new RsaEncryptionProvider(privateRsaKeyFile, null);
+        }
+        catch(IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)